### PR TITLE
Return non-zero exit status if failed to parse the action

### PR DESCRIPTION
**Pull request by gukoff**
### Issue describing the changes in this PR

Do not swallow `CliArgumentsException`. Return non-zero exit code if CLI arguments are incorrect.

Fixes https://github.com/Azure/azure-functions-core-tools/issues/3277

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by gukoff** [comment]
It likely needs backporting. What's the right procedure?

**Comment by gukoff** [comment]
ping!

File: test/Azure.Functions.Cli.Tests/ActionsTests/ResolveActionTests.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ public class ResolveActionTests
         [InlineData("azure functionapp enable-git-repo appName", typeof(DeprecatedAzureActions))]
         [InlineData("azure functionapp fetch-app-settings appName", typeof(FetchAppSettingsAction))]
         [InlineData("azure functionapp fetch appName", typeof(FetchAppSettingsAction))]
+        [InlineData("azure functionapp publish app-name -g resource-group", typeof(FetchAppSettingsAction))]
```
Comment (kshyju) [inline test/Azure.Functions.Cli.Tests/ActionsTests/ResolveActionTests.cs:25]: The test change is not validating the code change (the exit code is non zero). Can we add/update the tests do so?

**Comment by kshyju** [comment]
Thanks for your contribution @gukoff . Left one comment about the tests.

File: test/Azure.Functions.Cli.Tests/ActionsTests/ResolveActionTests.cs (hunk 1 above)
Comment (gukoff) [reply test/Azure.Functions.Cli.Tests/ActionsTests/ResolveActionTests.cs:25]: I added such a test below; do you mean something else?

**Review by aishwaryabh — APPROVED** [review]

**Review by kshyju — APPROVED** [review]

**Pull request merged**